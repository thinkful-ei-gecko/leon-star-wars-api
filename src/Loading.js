import React from 'react'
import LoadingLogo from './150.gif'
import './Loading.css'

export default function Loading() {
  return(
  <div className="loading">
    <img src={LoadingLogo} alt="loading" />
  </div>
  );
}