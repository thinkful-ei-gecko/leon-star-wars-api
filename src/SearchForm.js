import React, { Component } from 'react'

export default class SearchForm extends Component {
  state = {
    category: '',
    searchTerm: ''
  }

  changeCategory = (e) => {
    this.setState({
      category: e.target.value
    })
  }

  changeSearchTerm = (e) => {
    this.setState({
      searchTerm: e.target.value
    })
  }


  submitForm = (e) => {
    e.preventDefault();
    this.props.apiCall(this.state.category,this.state.searchTerm);
  }

  render() {

    let categories = this.props.categories || {};
    return (
      <form id="search-form" name="search-form" onSubmit={(e) => { this.submitForm(e)}
      }>
        <label htmlFor="category-select">Category:</label>
        <select id="category-select" name="category-select" onChange={(e) => this.changeCategory(e)} required>
            <option value="">Select one of the following:</option>
            {Object.keys(categories).map(category => <option value={category} key={category}>{category}</option> )}
          </select>
        <label htmlFor="search-input">Search Term:</label>
        <input type="text" placeholder="skywalker..." onChange={(e) => this.changeSearchTerm(e)} required/>
        <button type="submit" value="search!" className="useButton">search!</button>
      </form>
    )
  }
}