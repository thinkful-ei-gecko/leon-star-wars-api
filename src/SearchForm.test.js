import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchForm from './SearchForm'
import Loading from './Loading'

let categoriesList = [ 'test', 'data', 'here', 'okay'];

describe('<SearchForm />', () => {

  it('renders empty fine', () => {
    const wrapper = shallow(<SearchForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
    
  it('renders with categories in select', () => {
    const wrapper = shallow(<SearchForm categories={categoriesList} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('changing a select option calls the proper function', () => {
    const wrapper = shallow(<SearchForm categories={categoriesList} />)
    wrapper.find('select').at(0).simulate('change', {target: { value: 'testSelect'} })
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('changing the input calls the proper function', () => {
    const wrapper = shallow(<SearchForm categories={categoriesList} />)
    wrapper.find('input').at(0).simulate('change', {target: { value: 'testInput'} })
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('clicking submit submits the form', () => {
    const wrapper = shallow(<SearchForm categories={categoriesList} />)
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('submitting form calls proper function', () => {
    const wrapper = shallow(<SearchForm categories={categoriesList} apiCall={function() {}} />)
    wrapper.find('form').at(0).simulate('submit', { preventDefault: function() {}})
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

describe('<Loading />', () => {
  it('renders empty fine', () => {
    const wrapper = shallow(<Loading />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});