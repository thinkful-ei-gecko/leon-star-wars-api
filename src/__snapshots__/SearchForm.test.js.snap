// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders empty fine 1`] = `
<div
  className="loading"
>
  <img
    alt="loading"
    src="150.gif"
  />
</div>
`;

exports[`<SearchForm /> changing a select option calls the proper function 1`] = `
<form
  id="search-form"
  name="search-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="category-select"
  >
    Category:
  </label>
  <select
    id="category-select"
    name="category-select"
    onChange={[Function]}
    required={true}
  >
    <option
      value=""
    >
      Select one of the following:
    </option>
    <option
      key="0"
      value="0"
    >
      0
    </option>
    <option
      key="1"
      value="1"
    >
      1
    </option>
    <option
      key="2"
      value="2"
    >
      2
    </option>
    <option
      key="3"
      value="3"
    >
      3
    </option>
  </select>
  <label
    htmlFor="search-input"
  >
    Search Term:
  </label>
  <input
    onChange={[Function]}
    placeholder="skywalker..."
    required={true}
    type="text"
  />
  <button
    className="useButton"
    type="submit"
    value="search!"
  >
    search!
  </button>
</form>
`;

exports[`<SearchForm /> changing the input calls the proper function 1`] = `
<form
  id="search-form"
  name="search-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="category-select"
  >
    Category:
  </label>
  <select
    id="category-select"
    name="category-select"
    onChange={[Function]}
    required={true}
  >
    <option
      value=""
    >
      Select one of the following:
    </option>
    <option
      key="0"
      value="0"
    >
      0
    </option>
    <option
      key="1"
      value="1"
    >
      1
    </option>
    <option
      key="2"
      value="2"
    >
      2
    </option>
    <option
      key="3"
      value="3"
    >
      3
    </option>
  </select>
  <label
    htmlFor="search-input"
  >
    Search Term:
  </label>
  <input
    onChange={[Function]}
    placeholder="skywalker..."
    required={true}
    type="text"
  />
  <button
    className="useButton"
    type="submit"
    value="search!"
  >
    search!
  </button>
</form>
`;

exports[`<SearchForm /> clicking submit submits the form 1`] = `
<form
  id="search-form"
  name="search-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="category-select"
  >
    Category:
  </label>
  <select
    id="category-select"
    name="category-select"
    onChange={[Function]}
    required={true}
  >
    <option
      value=""
    >
      Select one of the following:
    </option>
    <option
      key="0"
      value="0"
    >
      0
    </option>
    <option
      key="1"
      value="1"
    >
      1
    </option>
    <option
      key="2"
      value="2"
    >
      2
    </option>
    <option
      key="3"
      value="3"
    >
      3
    </option>
  </select>
  <label
    htmlFor="search-input"
  >
    Search Term:
  </label>
  <input
    onChange={[Function]}
    placeholder="skywalker..."
    required={true}
    type="text"
  />
  <button
    className="useButton"
    type="submit"
    value="search!"
  >
    search!
  </button>
</form>
`;

exports[`<SearchForm /> renders empty fine 1`] = `
<form
  id="search-form"
  name="search-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="category-select"
  >
    Category:
  </label>
  <select
    id="category-select"
    name="category-select"
    onChange={[Function]}
    required={true}
  >
    <option
      value=""
    >
      Select one of the following:
    </option>
  </select>
  <label
    htmlFor="search-input"
  >
    Search Term:
  </label>
  <input
    onChange={[Function]}
    placeholder="skywalker..."
    required={true}
    type="text"
  />
  <button
    className="useButton"
    type="submit"
    value="search!"
  >
    search!
  </button>
</form>
`;

exports[`<SearchForm /> renders with categories in select 1`] = `
<form
  id="search-form"
  name="search-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="category-select"
  >
    Category:
  </label>
  <select
    id="category-select"
    name="category-select"
    onChange={[Function]}
    required={true}
  >
    <option
      value=""
    >
      Select one of the following:
    </option>
    <option
      key="0"
      value="0"
    >
      0
    </option>
    <option
      key="1"
      value="1"
    >
      1
    </option>
    <option
      key="2"
      value="2"
    >
      2
    </option>
    <option
      key="3"
      value="3"
    >
      3
    </option>
  </select>
  <label
    htmlFor="search-input"
  >
    Search Term:
  </label>
  <input
    onChange={[Function]}
    placeholder="skywalker..."
    required={true}
    type="text"
  />
  <button
    className="useButton"
    type="submit"
    value="search!"
  >
    search!
  </button>
</form>
`;

exports[`<SearchForm /> submitting form calls proper function 1`] = `
<form
  id="search-form"
  name="search-form"
  onSubmit={[Function]}
>
  <label
    htmlFor="category-select"
  >
    Category:
  </label>
  <select
    id="category-select"
    name="category-select"
    onChange={[Function]}
    required={true}
  >
    <option
      value=""
    >
      Select one of the following:
    </option>
    <option
      key="0"
      value="0"
    >
      0
    </option>
    <option
      key="1"
      value="1"
    >
      1
    </option>
    <option
      key="2"
      value="2"
    >
      2
    </option>
    <option
      key="3"
      value="3"
    >
      3
    </option>
  </select>
  <label
    htmlFor="search-input"
  >
    Search Term:
  </label>
  <input
    onChange={[Function]}
    placeholder="skywalker..."
    required={true}
    type="text"
  />
  <button
    className="useButton"
    type="submit"
    value="search!"
  >
    search!
  </button>
</form>
`;
